exporter/containerimage/writer.go

```diff
@@ -333,6 +385,132 @@ func (ic *ImageWriter) commitDistributionManifest(ctx context.Context, ref cache
 	}, &configDesc, nil
 }
 
+func (ic *ImageWriter) commitAttestationsManifest(ctx context.Context, p exptypes.Platform, target string, oci bool, attestations []exporter.Attestation, inTotos []InTotoAttestation) (*ocispecs.Descriptor, *ocispecs.Descriptor, error) {
+	var (
+		attestationType                = exptypes.MediaTypeAttestationType
+		manifestType                   = ocispecs.MediaTypeImageManifest
+		configType                     = ocispecs.MediaTypeImageConfig
+		annotationReferenceType        = exptypes.OCIAnnotationReferenceType
+		annotationReferenceDigest      = exptypes.OCIAnnotationReferenceDigest
+		annotationReferenceDescription = exptypes.OCIAnnotationReferenceDescription
+	)
+	if !oci {
+		attestationType = exptypes.MediaTypeDockerSchema2AttestationType
+		manifestType = images.MediaTypeDockerSchema2Manifest
+		configType = images.MediaTypeDockerSchema2Config
+		annotationReferenceType = exptypes.DockerAnnotationReferenceType
+		annotationReferenceDigest = exptypes.DockerAnnotationReferenceDigest
+		annotationReferenceDescription = exptypes.DockerAnnotationReferenceDescription
+	}
+
+	layers := []ocispecs.Descriptor{}
+	for _, inToto := range inTotos {
+		data, err := json.Marshal(inToto)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		digest := digest.FromBytes(data)
+		desc := ocispecs.Descriptor{
+			MediaType: attestationType,
+			Digest:    digest,
+			Size:      int64(len(data)),
+			Annotations: map[string]string{
+				"containerd.io/uncompressed": digest.String(),
+				"in-toto.io/predicate-type":  inToto.PredicateType,
+			},
+		}
+		done := oneOffProgress(ctx, "exporting attestation "+digest.String())
+		if err := content.WriteBlob(ctx, ic.opt.ContentStore, digest.String(), bytes.NewReader(data), desc); err != nil {
+			return nil, nil, done(errors.Wrapf(err, "error writing data blob %s", digest))
+		}
+		done(nil)
+
+		layers = append(layers, desc)
+	}
+
+	config, err := emptyImageConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	history := createAttestationsHistory(ctx, layers, attestations)
+	config, err = patchImageConfig(config, layers, history, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	configDigest := digest.FromBytes(config)
+
+	mfst := struct {
```

Comment:
I think you use `man` in another place... I tend to prefere `manifest`, real words make good variable names :)

Reply:
Going to go for `mfst` in this file, to keep consistent with the rest of the containerimage exporter, and to avoid making this PR larger and more difficult to review.